core/src/ast/modules.rs

```diff
@@ -130,12 +137,36 @@ impl Module {
 
         Module {
             name: input.ident.to_string(),
+            imports,
             declared_types: custom_types_by_name,
             sub_modules,
         }
     }
 }
 
+fn extract_imports(base_path: &Path, use_tree: &UseTree, out: &mut Vec<(Path, String)>) {
```

Comment:
I would only extract tagged imports, other imports should just be ignored IMO.

Reply:
Hmm, is there any advantage from the Diplomat side for doing this? We handle unsupported imports by just not adding symbols at resolution time anyways, so was thinking of only requiring annotations for DSTs to reduce noise.